HPXMLtoOpenStudio/resources/EPvalidator.xml

```diff
@@ -139,12 +152,22 @@
       <sch:assert role='ERROR' test='count(h:ConditionedBuildingVolume) + count(h:AverageCeilingHeight) &gt;= 1'>Expected 1 or more element(s) for xpath: ConditionedBuildingVolume | AverageCeilingHeight</sch:assert>
       <sch:assert role='ERROR' test='count(h:extension/h:HasFlueOrChimney) &lt;= 1'>Expected 0 or 1 element(s) for xpath: extension/HasFlueOrChimney</sch:assert>
     </sch:rule>
+    <!-- FIXME: Does the following rules need to be wrapped in separate patterns for consistency? -->
+    <!-- [ResidentialFacilityType] -->
+    <sch:rule context='/h:HPXML/h:Building/h:BuildingDetails/h:BuildingSummary/h:BuildingConstruction/h:ResidentialFacilityType'>
+      <sch:assert role='ERROR' test='contains("_single-family detached_ _single-family attached_ _apartment unit_ _manufactured home_", concat("_", text(), "_"))'>Expected value to be 'single-family detached' or 'single-family attached' or 'apartment unit' or 'manufactured home'</sch:assert>
+    </sch:rule>
```

Comment:
@bpark1327 Do you think we can go the same approach we went for HPXMLvalidator.xml? E.g., remove the additional rule, and change the assert to be:
```
<sch:assert role='ERROR' test='not(h:ResidentialFacilityType) or contains("_single-family detached_ _single-family attached_ _apartment unit_ _manufactured home_", concat("_", h:ResidentialFacilityType, "_"))'>Expected ResidentialFacilityType to be 'single-family detached' or 'single-family attached' or 'apartment unit' or 'manufactured home'</sch:assert>
```

See https://github.com/NREL/OpenStudio-HPXML/blob/master/HPXMLtoOpenStudio/resources/HPXMLvalidator.xml#L22 for corresponding assert in HPXMLvalidator.xml.

Reply:
If we want to use the same approach, the test will be slightly different depending on whether an element is optional or required. 
For optional elements,
`<sch:assert role='ERROR' test='not(h:Foo) or contains("_Bar_ _Baz_", concat("_", h:Foo, "_"))'>Expected Foo to be 'Bar' or 'Baz'</sch:assert>`
For required elements,
`<sch:assert role='ERROR' test='count(h:Foo) = 1 and contains("_Bar_ _Baz_", concat("_", h:Foo, "_"))'>Expected Foo to be 'Bar' or 'Baz'</sch:assert>`
I'm not sure what would be the best way to distinguish between them in messages. Perhaps, we should keep the expected number of elements in messages (e.g., Expected 0 or 1 Foo to be 'Bar' or 'Baz').

Depending on how we want to change our messages, we may need to think about how we want to assign each assertion to different validation tests (i.e. addition, deletion, and alteration tests).

Reply:
I'm not suggesting that we recombine the two tests back into one, just that we change the new test you're creating.

So an optional element becomes:

```
<sch:assert role='ERROR' test='Foo &lt;=1'>Expected 0 or 1 element(s) for xpath: Foo</sch:assert>
<sch:assert role='ERROR' test='not(h:Foo) or contains("_Bar_ _Baz_", concat("_", h:Foo, "_"))'>Expected Foo to be 'Bar' or 'Baz'</sch:assert>
```

If they provide two Foo's, they get the first error. If they provide one Foo but it has the wrong value, they get the second error.

And a required element becomes:

```
<sch:assert role='ERROR' test='Foo =1'>Expected 1 element(s) for xpath: Foo</sch:assert>
<sch:assert role='ERROR' test='not(h:Foo) or contains("_Bar_ _Baz_", concat("_", h:Foo, "_"))'>Expected Foo to be 'Bar' or 'Baz'</sch:assert>
```

If they provide zero Foo's, they get the first error. If they provide one Foo but it has the wrong value, they get the second error.

Reply:
(In other words, the test is the same for optional and required elements.)

Reply:
Aha, that makes sense! I'll pick this up. Thanks for the clarification.